### DISCUSSION TITLE
Extension | Useful Chrome's API Support | Built-in usercript manager | Environmental Variables

### DISCUSSION THREAD

**Discussion by bachig26** [FreeTubeApp/FreeTube#2021] [General]
As I went across the Electron today, I found that a subset of the Chrome Extensions API can be supported by the Electron Framework.
I think if these APIs are to be implemented within FreeTube, it will be fun to work on.
I believe we need to review the useful Chrome APIs to strip down the Analytics, telemetry, or any others that need to be removed.

[1] https://www.electronjs.org/docs/latest/api/extensions

[2] https://developer.chrome.com/docs/extensions/reference/

From API Reference [2], I found these are useful and will be used by many users,
APIs like, 
- [downloads](https://developer.chrome.com/docs/extensions/reference/downloads)
- [tabs](https://developer.chrome.com/docs/extensions/reference/tabs), [tabGroups](https://developer.chrome.com/docs/extensions/reference/tabGroups)
- [extension](https://developer.chrome.com/docs/extensions/reference/extension)
- and others.

I believe that there needs to be a **usercript manager within the FreeTube,** as there are many YouTube-related open Source userscripts that we can take advantage of.

Also, I wanted to ask if the FreeTube can be made use of the "**Environmental Variables**" to control, configure FreeTube and add more functionality?
https://www.electronjs.org/docs/latest/api/environment-variables

**Comment by PrestonN** [comment]
Just because there are a lot of userscripts out there for YouTube doesn't mean that they would work for FreeTube. Most of them would have to be rebuilt most likely and at that point it's probably better if the writer were to make a PR to the app instead. We can consider it much further down the line if we feel it's needed, but I don't see this as a priority at this time.
